test/features.jl

```diff
@@ -449,12 +449,12 @@ end
 @test pullback(type_test)[1] == Complex{<:Real}
 
 @testset "Pairs" begin
-    @test (x->10*pairs((a=x, b=2))[1])'(100) === 10
+    @test (x->10*pairs((a=x, b=2))[1])'(100) === 10.0
```

Comment:
this is because we now take embedded sub-spaces seriously.
Integers are always considered to be a subspace of Floats.
Not just went it is convenient.

Reply:
Shouldn't this be compiling to integer code anyway?

Reply:
Integers are not a good type to use to represent tangents.
Because if you are going to do gradeient decent you are going to apply a learning rate like `0.1*dx`.
So we call `float` on them to get the corresponding floating point type.
(Unless it is a index or something, then it would be `NoTangent()`)

Arguably we could keep integers here, but encouraging people to use integers to repressent continous values that just so happen do be integers feels like not taking subspace types seriously
:shrug:

Reply:
I think it's a matter of letting the language take charge of these things. It's possible to work with just ints.

Reply:
Can this be moved to a issue on ChainRulesCore.jl?
Its not going to be resolved in this PR.